pkg/pgmodel/ingestor/trace/tag_batch.go

```diff
@@ -94,34 +94,34 @@ func (t tagBatch) SendBatch(ctx context.Context, conn pgxconn.PgxConn) (err erro
 	return t.b.SendBatch(ctx, conn)
 }
 
-func (tb tagBatch) GetTagMapJSON(tags map[string]interface{}, typ TagType) ([]byte, error) {
+func (tb tagBatch) GetTagMapJSON(tags map[string]interface{}, typ TagType) (pgtype.JSONB, error) {
 	tagMap := make(map[int64]int64)
 	for k, v := range tags {
 		byteVal, err := json.Marshal(v)
 		if err != nil {
-			return nil, err
+			return pgtype.JSONB{}, err
 		}
 		t := tag{k, string(byteVal), typ}
 		ids, err := tb.b.Get(t)
 		if err != nil {
-			return nil, fmt.Errorf("error getting tag from batch %v: %w", t, err)
+			return pgtype.JSONB{}, fmt.Errorf("error getting tag from batch %v: %w", t, err)
```

Comment:
Imho it makes more sense to return `nil` for error cases.

Reply:
we can't since the return type is not a pointer